raiden/network/proxies/secret_registry.py

```diff
@@ -59,23 +70,44 @@ def register_secret_batch(
         secrets_to_register = list()
         secrethashes_to_register = list()
         secrethashes_not_sent = list()
-        secret_registry_transaction = AsyncResult()
+        transaction_result = AsyncResult()
 
-        for secret in secrets:
-            secrethash = sha3(secret)
-            secrethash_hex = encode_hex(secrethash)
+        with self._open_secret_transactions_lock:
+            for secret in secrets:
+                secrethash = sha3(secret)
+                secrethash_hex = encode_hex(secrethash)
 
-            is_register_needed = (
-                not self.check_registered(secrethash, given_block_identifier) and
-                secret not in self.open_secret_transactions
-            )
-            if is_register_needed:
-                secrets_to_register.append(secret)
-                secrethashes_to_register.append(secrethash_hex)
-                self.open_secret_transactions[secret] = secret_registry_transaction
```

Comment:
This was a bug, without synchronization `check_registered` above context switches and different threads may try to register the same secret